### PR TITLE
Fix the issue in refresh grant flow with sso-binding

### PR DESCRIPTION
**Pull request by piraveena**
### Proposed changes in this pull request

Resolves https://github.com/wso2/product-is/issues/11247

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/impl/AbstractTokenBinder.java [hunk 1]
```diff
@@ -105,8 +108,12 @@ private boolean isValidTokenBinding(OAuth2AccessTokenReqDTO oAuth2AccessTokenReq
                     String cookiePrefix = cookieName + "=";
                     for (String cookie : cookies) {
                         if (StringUtils.isNotBlank(cookie) && cookie.trim().startsWith(cookiePrefix)) {
-                            String receivedBindingReference = OAuth2Util
-                                    .getTokenBindingReference(HttpCookie.parse(cookie).get(0).getValue());
+                            String tokenBindingValue = HttpCookie.parse(cookie).get(0).getValue();
```
Comment (IsuraD) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/impl/AbstractTokenBinder.java:111]: Better do null check HttpCookie.parse(cookie).get(0)

Comment (piraveena) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/impl/AbstractTokenBinder.java:112]: fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1547/commits/c049aea8e3a971f72dcba39f274951ab5848a07b

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/impl/AbstractTokenBinder.java [hunk 2]
```diff
@@ -43,6 +44,8 @@
  */
 public abstract class AbstractTokenBinder implements TokenBinder {
 
+    private static final String COMMONAUTH_COOKIE = "commonAuthId";
```
Comment (mevan-karu) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/impl/AbstractTokenBinder.java:47]: There should be a constant defined for this in the identity-framework. Better to reuse that.

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/impl/AbstractTokenBinder.java [hunk 3]
```diff
@@ -104,10 +107,21 @@ private boolean isValidTokenBinding(OAuth2AccessTokenReqDTO oAuth2AccessTokenReq
...
-                        if (StringUtils.isNotBlank(cookie) && cookie.trim().startsWith(cookiePrefix)) {
-                            String receivedBindingReference = OAuth2Util
-                                    .getTokenBindingReference(HttpCookie.parse(cookie).get(0).getValue());
-                            return bindingReference.equals(receivedBindingReference);
+                        if (StringUtils.isNotBlank(cookie) && cookie.trim().startsWith(cookiePrefix) &&
+                                HttpCookie.parse(cookie).get(0) != null) {
+                            String cookieValue = HttpCookie.parse(cookie).get(0).getValue();
+                            if (StringUtils.isNotBlank(cookieValue)) {
+                                String tokenBindingValue;
+                                if (cookieName.equals(COMMONAUTH_COOKIE)) {
```
Comment (mevan-karu) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/impl/AbstractTokenBinder.java:115]:
```suggestion
                                if (COMMONAUTH_COOKIE.equals(cookieName)) {
```

**Review by mevan-karu — APPROVED** [review]

**Pull request merged**